akka-http/src/main/scala/akka/http/javadsl/server/directives/FileUploadDirectives.scala

```diff
@@ -34,6 +37,19 @@ abstract class FileUploadDirectives extends FileAndResourceDirectives {
   def fileUpload(fieldName: String, inner: BiFunction[FileInfo, Source[ByteString, Any], Route]): Route = RouteAdapter {
     D.fileUpload(fieldName) { case (info, src) ⇒ inner.apply(info, src.asJava).delegate }
   }
+
+  /**
+   * Collects each body part that is a multipart file as a tuple containing metadata and a `Source`
+   * for streaming the file contents somewhere. If there is no such field the request will be rejected.
+   */
+  def fileMultiUpload(fieldName: String, inner: JFunction[JList[JMap.Entry[FileInfo, Source[ByteString, Any]]], Route]): Route = RouteAdapter {
+    def toJava[F <: FileInfo](info: F): FileInfo = info
```

Comment:
I'm not super happy about this bit, but without it I'd need to `asInstanceOf` to make it work with `Map.Entry`.